src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullCoalescingOperator.cs

```diff
@@ -141,6 +141,18 @@ public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperato
                 return BoundCall.Synthesized(rewrittenLeft.Syntax, rewrittenLeft, getValueOrDefault);
             }
 
+            // Optimize `left ?? true` to `left.GetValueOrDefault() | !left.HasValue`.
+            if (rewrittenRight.ConstantValueOpt == ConstantValue.True
```

Comment:
I've implemented more optimal `left.GetValueOrDefault(true)` lowering for this case in https://github.com/dotnet/roslyn/pull/70199. It is more generic (applies not only to booleans), results in smaller IL thus is very good for JIT inlining. I think, we want to get #70199 first and then adjust this PR based on that